Fix: to fix SMTrainingCompilerConfigurationError handling in process.py

vinayburugu opened the pull request:
*Issue #, if available:*

*Description of changes:*
The code in process.py check_error assumes all errors in errors.py as sub classes of _CalledProcessError and reaises that error. Handled SMTrainingCompilerConfigurationError differently as it is not subclass of _CalledProcessError. Reverted the code of registering SMTrainingCompilerConfigurationError with toolkit ([PR](https://github.com/aws/sagemaker-training-toolkit/pull/171/files)) as SMTrainingCompilerConfigurationError can only be raised by toolkit and not the user training script.


*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-training-toolkit/blob/master/CONTRIBUTING.md) doc
- [ x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-training-toolkit/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have used the regional endpoint when creating S3 and/or STS clients (if appropriate)
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-training-toolkit/blob/master/README.md)

#### Tests

- [ x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Lokiiiiii commented on src/sagemaker_training/process.py:
```diff
@@ -325,7 +325,14 @@ def check_error(cmd, error_classes, processes_per_host, cwd=None, capture_error=
+                    {},
+                )
+                raise error_class(stderr)
+            else:
```
If error_name is empty, this logic will fail.

satishpasumarthi commented:
@vinayburugu  There are a lot of formatting changes in this PR which makes it harder to review. Are you running any other formatting ?

Also, I see that you are undoing a lot of changes. could you please point to any relevant PR if you are reverting those changes. It will help us review better.

vinayburugu commented:
@satishpasumarthi The formatted changes are unintended. I ran black -l 100 on the files I modified. I basically reverted the [PR](https://github.com/aws/sagemaker-training-toolkit/pull/171/files) . In this PR, I removed registering the keyword SMTrainingCompilerConfigurationError and handled the error explicitly.

vinayburugu commented:
> @satishpasumarthi The formatted changes are unintended. I ran black -l 100 on the files I modified. I basically reverted the [PR](https://github.com/aws/sagemaker-training-toolkit/pull/171/files) . In this PR, I removed registering the keyword SMTrainingCompilerConfigurationError and handled the error explicitly.

yl-to commented:
Had similar concerns as @satishpasumarthi 
1. Can we revert back unnecessary format changes?
2. Can you describe the reason why we add compiler errors in previous PR and now removing it in PR notes?

Thanks!
@vinayburugu

yl-to commented on src/sagemaker_training/process.py:
```diff
@@ -326,6 +317,15 @@ def check_error(cmd, error_classes, processes_per_host, cwd=None, capture_error=
+                        ),  # pylint: disable=protected-access
+                        {},
+                    )
+                    raise error_class(stderr)
```
How to make sure this error will be raised after we removed the test?

yl-to replied:
Can you elaborate the functionality of these added lines?

yl-to replied:
since the previous test is removed, can we have a simple test to test catching this error?

vinayburugu replied:
added the test.
